iwharfclientfetcher.go

```diff
@@ -1,14 +1,38 @@
 package main
 
-import "github.com/iver-wharf/wharf-api-client-go/pkg/wharfapi"
+import (
+	"github.com/iver-wharf/wharf-api-client-go/v2/pkg/model/request"
+	"github.com/iver-wharf/wharf-api-client-go/v2/pkg/model/response"
+	"github.com/iver-wharf/wharf-api-client-go/v2/pkg/wharfapi"
+)
 
 type wharfClientAPIFetcher interface {
-	GetTokenByID(tokenID uint) (wharfapi.Token, error)
-	GetToken(token string, userName string) (wharfapi.Token, error)
-	PostToken(token wharfapi.Token) (wharfapi.Token, error)
-	GetProviderByID(providerID uint) (wharfapi.Provider, error)
-	GetProvider(providerName string, urlStr string, uploadURLStr string, tokenID uint) (wharfapi.Provider, error)
-	PostProvider(provider wharfapi.Provider) (wharfapi.Provider, error)
-	PutProject(project wharfapi.Project) (wharfapi.Project, error)
-	PutBranches(branches []wharfapi.Branch) ([]wharfapi.Branch, error)
+	GetBuildArtifactList(params wharfapi.ArtifactSearch, buildID uint) (response.PaginatedArtifacts, error)
+	GetBuildArtifact(buildID, artifactID uint) (response.Artifact, error)
+	CreateProjectBranch(projectID uint, branch request.Branch) (response.Branch, error)
+	UpdateProjectBranchList(projectID uint, branches []request.Branch) ([]response.Branch, error)
+	GetProjectBranchList(projectID uint) ([]response.Branch, error)
+	GetBuildList(params wharfapi.BuildSearch) (response.PaginatedBuilds, error)
+	GetBuild(buildID uint) (response.Build, error)
+	UpdateBuildStatus(buildID uint, status request.LogOrStatusUpdate) (response.Build, error)
+	CreateBuildLog(buildID uint, buildLog request.LogOrStatusUpdate) error
```

Comment:
This interface should only need to include the methods it's actually going to use. Redeclaring all methods here is a bit excessive.

For example, the wharf-provider-gitlab will never create build logs, so no need to include that.

Reply:
Pruned it to only have ones in use and a few that I think might be required to support refreshing projects again (which I noticed just now will not work currently :o )